src/main/java/formflow/library/file/FileConversionService.java

```diff
@@ -386,6 +395,61 @@ private Set<MultipartFile> convertOfficeDocumentToPDF(MultipartFile file, int nu
         }
     }
 
+    /**
+     * Takes a multipartfile representation of a PDF and makes a clone of it, with the relax security settings A read-only PDF is
+     * now not as read-only
+     *
+     * @param file
+     * @return
+     */
+    private Set<MultipartFile> relaxPDFSecuritySettings(MultipartFile file) {
+
+        // Write to a temp file, so we can have a File from the original MultipartFile
+        String originalFilename = file.getOriginalFilename();
+        if (originalFilename == null || originalFilename.contains("..") || originalFilename.contains("/")
+                || originalFilename.contains("\\")) {
+            throw new IllegalArgumentException("Unable to relax security setting for PDF. Invalid filename.");
+        }
+
+        File tempFile = null;
+        try {
+            tempFile = File.createTempFile("upload_", "_" + file.getOriginalFilename());
+
+            try (FileOutputStream fos = new FileOutputStream(tempFile)) {
+                fos.write(file.getBytes());
+            }
+
+            File modifiedPDFFile = getModifiedPDFFile(tempFile);
```

Comment:
I might be confused, but are we calling `getModifiedPDFFile` before renaming it, thus the file name wouldn't be mutated to include `_modified` yet, causing `getModifiedPDFFile` not to find anything?

Reply:
Or maybe it's that getModifiedPDFFile should be looking for upload_originalFileName instead of tempFile.getAbsolutePath()_modified.pdf ?